src/hooks/useGraphQLClient.ts

```diff
@@ -11,46 +11,107 @@ import { ApolloClient } from '@apollo/client/core/ApolloClient';
 import { setContext } from '@apollo/client/link/context';
 import { onError } from '@apollo/client/link/error';
 import { RetryLink } from '@apollo/client/link/retry';
+import { useMemo, useRef } from 'react';
 import { useDispatch } from 'react-redux';
 import { env } from '../env';
 import { typePolicies } from '../graphql/cache/typePolicies';
+import { ErrorStatus } from '../models/Errors';
+import { updateStatus, updateToken } from '../reducers/auth/actions';
+import { AuthStatus } from '../reducers/auth/types';
 import { pushError } from '../reducers/error/actions';
-import { useAuthenticate } from './useAuthenticate';
-import { TOKEN_STORAGE_KEY } from './useAuthentication';
-import { updateStatus } from '../reducers/auth/actions';
+import { useAuthenticationContext } from './useAuthenticationContext';
 
 const enableQueryDebug = !!(env && env?.REACT_APP_DEBUG_QUERY === 'true');
 
 export const useGraphQLClient = (graphQLEndpoint: string): ApolloClient<NormalizedCacheObject> => {
   const dispatch = useDispatch();
-  const { safeRefresh, status } = useAuthenticate();
-  let token: string | null;
+  const { context, status: _status, token: _token } = useAuthenticationContext();
+
+  // Preserve the token and status from the reduxStore to be used inside the memoized apollo client
+  const status = useRef<AuthStatus>('unauthenticated');
+  const token = useRef<string | undefined>();
+  token.current = _token;
+  status.current = _status;
+
+  const pendingRequests = useRef<((token?: string) => void)[]>([]);
+  const isRefreshing = useRef(false);
+  const counter = useRef(0);
+  const refreshCounter = useRef(0);
```

Comment:
i dont see the usage of it. seeing only the increment on line 47

Reply:
Oh yes, they were just for development purpose.